packages/compiler/core/types.ts

```diff
@@ -18,10 +18,16 @@ export type DecoratorArgumentValue = Type | number | string | boolean;
 
 export interface DecoratorArgument {
   value: Type;
+  /**
+   * Marshalled value for use in Javascript.
+   */
+  jsValue: Type | string | number | boolean;
```

Comment:
in the presence of object types, it seems like it wouldn't be possible to discriminate betweeh `Type` and some object that happens to have the same shape. Perhaps we need to add an envelope here?

Reply:
like if passing 
```
@dec(#{
  kind: "Model"
})
```

feels like there is no good way to not have this be breaking or we need to make the object literal be in an envolope all the time(which basically is pretty much the type

Reply:
Yeah that's the scenario. But I don't think it needs to be an envelope outside the compiler internals? When passing to a decorator we wouldn't need to pass in the envelope.

Reply:
Ah I see, the only place I can see useful here is in the projector where we decide to proejct or not the type if its a `Type` or a value. But there we could reresolve the `jsValue` from the projected `value`